### PR TITLE
Numerics serialization

### PR DESCRIPTION
**Pull request by morganbr**
Adds serialization support to BigInteger and Complex, which were inadvertently dropped in the previous cleanup. Also renames the private fields of Complex to match NetFX so that it will be serialization compatible. Testing will be covered in an upcoming change by @ViktorHofer.

This is progress on #19119

### CODE REVIEW

**Comment by morganbr** [comment]
CC @danmosemsft @dotnet/corert-contrib

File: src/System.Runtime.Numerics/src/System/Numerics/Complex.cs [hunk 1]
```diff
@@ -29,20 +30,20 @@ public struct Complex : IEquatable<Complex>, IFormattable
...
-        private double _imaginary;
+        private double m_real;
+        private double m_imaginary;
         
         public Complex(double real, double imaginary)
         {
-            _real = real;
-            _imaginary = imaginary;
+            m_real = real;
+            m_imaginary = imaginary;
```
Comment (danmosemsft) [inline src/System.Runtime.Numerics/src/System/Numerics/Complex.cs]: Are we going to commentthat field names must not change, ornot

Comment (sharwell) [inline src/System.Runtime.Numerics/src/System/Numerics/Complex.cs]: Assuming the upcoming test covers it, I wouldn't think the comment would be necessary. Seems like adding the comments is prone to a situation where one or more comments is omitted, and a reader incorrectly infers that the lack of a comment means the name can change.

Comment (stephentoub) [inline src/System.Runtime.Numerics/src/System/Numerics/Complex.cs]: I would prefer we add comments.  I understand @sharwell's point, but at the same time, seeing a comment and avoiding a change entirely is a lot quicker than having to figure out why a test is failing, can be spotted in code reviews, etc.

Comment (danmosemsft) [inline src/System.Runtime.Numerics/src/System/Numerics/Complex.cs]: Maybe something inline like 
```c#
            m_real = real; // Do not rename (binary serialization)
            m_imaginary = imaginary; // Do not rename (binary serialization)
```

We already do stuff like this in types that are hard bound to the runtime eg on string and array, although I guess there will be many more like this.
Your call @morganbr

Comment (morganbr) [inline src/System.Runtime.Numerics/src/System/Numerics/Complex.cs]: I'll go ahead with those comments in this change, but leave the overall process to @ViktorHofer and @krwq when they handle all of the other types.

**Pull request merged**